R/recodeGADS.R

```diff
@@ -45,54 +45,63 @@
-      GADSdat$dat[is.na(GADSdat$dat[, varName]), varName] <- newValues[i]
-    } else {
-      changeTable[changeTable$varName == varName &
-                    !is.na(changeTable$value) & changeTable$value == oldValues[i], "value_new"] <- newValues[i]
+
+  for(single_varName in varName) {
+    for(i in seq_along(oldValues)) {
+      if(is.na(oldValues[i])) {
+        GADSdat$dat[is.na(GADSdat$dat[, single_varName]), single_varName] <- newValues[i]
+      } else {
+        changeTable[changeTable$varName == single_varName &
+                      !is.na(changeTable$value) & changeTable$value == oldValues[i], "value_new"] <- newValues[i]
+      }
+    }
+    # recode values without labels (not the best solution but better usability)
+    other_recodes <- which(!oldValues %in% changeTable[changeTable$varName == single_varName, "value"] & !is.na(oldValues))
+    for(i in other_recodes) {
+      if(!oldValues[i] %in% GADSdat$dat[, single_varName]) {
+        warning("The following value in 'oldValues' is neither a labeled value in the meta data nor an actual value in ",
+                single_varName, ": ", oldValues[i])
+      }
+      GADSdat$dat[which(GADSdat$dat[, single_varName] == oldValues[i]), single_varName] <- newValues[i]
     }
   }
-  out <- applyChangeMeta(GADSdat, changeTable = changeTable, existingMeta = existingMeta)
 
-  # recode values without labels (not the best solution but better usability)
-  other_recodes <- which(!oldValues %in% changeTable[changeTable$varName == varName, "value"] & !is.na(oldValues))
-  for(i in other_recodes) {
-    if(!oldValues[i] %in% GADSdat$dat[, varName]) warning("The following value in 'oldValues' is neither a labeled value in the meta data nor an actual value in ",
-                                                          varName, ": ", oldValues[i])
-    out$dat[which(GADSdat$dat[, varName] == oldValues[i]), varName] <- newValues[i]
-  }
-  out
+  applyChangeMeta(GADSdat, changeTable = changeTable, existingMeta = existingMeta)
 }
 
 #'@export
 recodeGADS.all_GADSdat <- function(GADSdat, varName, oldValues, newValues, existingMeta = c("stop", "value", "value_new", "drop", "ignore")) {
   check_all_GADSdat(GADSdat)
+
   singleGADS_list <- lapply(names(GADSdat$datList), function(nam ) {
     singleGADS <- extractGADSdat(GADSdat, name = nam)
-    if(varName %in% names(singleGADS$dat)) singleGADS <- recodeGADS(singleGADS, varName = varName, oldValues = oldValues,
-                                                                    newValues = newValues, existingMeta = existingMeta)
+    specific_varNames <- varName[varName %in% names(singleGADS$dat)]
+    if(length(specific_varNames) > 0) {
+      singleGADS <- recodeGADS(singleGADS, varName = specific_varNames, oldValues = oldValues,
+                              newValues = newValues, existingMeta = existingMeta)
+    }
     singleGADS
   })
   names(singleGADS_list) <- names(GADSdat$datList)
   do.call(mergeLabels, singleGADS_list)
 }

```

Comment:
Nit: Should there be an additional check for the existence of the specified variable(s) in the `GADSdat`'s, or would this eat up more resources than it saves and more than is worth for the improved user-friendliness? I guess there are two slightly different things that could be checked:
 1. Is *at least one* of the specified variables *actually* a variable in *any* of the `GADSdat`? Or will the function return exactly the same `all_GADSdat` it received?
 2. Is *any* of the specified variables *not actually* a variable in *any* of the `GADSdat`? Or was there maybe a typo on this specific variable name while the other variables will/would be recoded as requested?

Reply:
I agree, this is indeed inconsistent. I have added the same test (`check_vars_in_GADSdat()`) to `recodeGADS.all_GADSdat()` as well.

Thanks for spotting this!

Side note: 
`all_GADSdat` objects are currently used only in very specific circumstances in practice (probably because there is very limited information on them and the documentation leaves them out completey). Therefore, only a few functions support them.